loader/paths.go

```diff
@@ -68,13 +68,9 @@ func ResolveRelativePaths(project *types.Project) error {
 }
 
 func ResolveServiceRelativePaths(workingDir string, s *types.ServiceConfig) {
-	if s.Build != nil && s.Build.Context != "" && !isRemoteContext(s.Build.Context) {
-		// Build context might be a remote http/git context. Unfortunately supported "remote"
-		// syntax is highly ambiguous in moby/moby and not defined by compose-spec,
-		// so let's assume runtime will check
-		localContext := absPath(workingDir, s.Build.Context)
-		if _, err := os.Stat(localContext); err == nil {
-			s.Build.Context = localContext
+	if s.Build != nil {
+		if !isRemoteContext(s.Build.Context) {
+			s.Build.Context = absPath(workingDir, s.Build.Context)
 		}
 		for name, path := range s.Build.AdditionalContexts {
```

Comment:
N.B. There's actually an additional fix here for `build.additional_contexts` -- they were only being resolved if the full condition was met: `s.Build != nil && s.Build.Context != "" && !isRemoteContext(s.Build.Context)`